pkg/schemaconv/smd.go

```diff
@@ -122,32 +162,34 @@ func (c *convert) VisitArray(a *proto.Array) {
 		} else if val == "map" {
 			l.ElementRelationship = schema.Associative
 			if keys, ok := ext["x-kubernetes-list-map-keys"]; ok {
-				if keyNames, ok := keys.([]string); ok {
+				if keyNames, ok := toStringSlice(keys); ok {
 					l.Keys = keyNames
 				} else {
-					// emit warning
+					c.reportError("uninterpreted map keys: %#v", keys)
 				}
 			} else {
-				// emit warning
+				c.reportError("missing map keys")
 			}
 		} else {
-			// TODO: emit warning?
+			c.reportError("unknown list type %v", val)
 			l.ElementRelationship = schema.Atomic
 		}
 	} else if val, ok := ext["x-kubernetes-patch-strategy"]; ok {
-		if val == "merge" {
+		if val == "merge" || val == "merge,retainKeys" {
```

Comment:
You could maybe handle retainsKeys correctly and transform it into something useful